src/app/ui/user-profile/user-profile.component.ts

```diff
@@ -1,17 +1,93 @@
-import { Component } from '@angular/core';
-
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { AuthService } from '../../core/auth.service';
+import { UserService } from '../../core/user.service';
+import { MatDialog } from '@angular/material/dialog';
+import { AddQuestDialogComponent } from '../dialog/add-quest-dialog/add-quest-dialog.component';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { PlayerQuestService } from '../player-quest/player-quest.service';
+import { SeasonService } from 'src/app/core/season.service';
+import { EmailService } from 'src/app/core/email.service';
+import { NotifyService } from 'src/app/core/notify.service';
 
 @Component({
   selector: 'user-profile',
   templateUrl: './user-profile.component.html',
   styleUrls: ['./user-profile.component.scss'],
 })
-export class UserProfileComponent {
+export class UserProfileComponent implements OnInit, OnDestroy {
+  private player: User;
+  private userServiceSub: Subscription;
+  private user: User;
+  private userSub: Subscription;
+  private seasonSub: Subscription;
 
-  constructor(public auth: AuthService) { }
+  constructor(public auth: AuthService,
+    private userService: UserService,
+    private dialog: MatDialog,
+    private route: ActivatedRoute,
+    private playerQuestService: PlayerQuestService,
+    private seasonService: SeasonService,
+    private emailService: EmailService,
+    private notifyService: NotifyService) { }
 
   logout() {
     this.auth.signOut();
   }
+
+  ngOnInit() {
+    // get the logged-in user data
+    this.userSub = this.auth.user.subscribe((user: User) => {
+      this.user = user;
+      if (this.user.team.lead === this.user.uid) {
```

Comment:
no need for this here, i'll fix the security rules for getUser to allow only owner and team lead for read access.
It will throw an error if it fails, so you need to catch the observable exception and just show a simple placeholder that the user is not allowed access to this profile.